test/dynamo/test_optimizers.py

```diff
@@ -44,10 +44,8 @@ def fn():
 
 
 def make_test(optim_cls, closure=None, **kwargs):
-    opt = optim_cls(model.parameters(), **kwargs)
-
     def test_fn(self):
-        nonlocal opt
+        opt = optim_cls(model.parameters(), **kwargs)
```

Comment:
@tfsingh could you test if this patch is still necessary after adding the AdamW exclusion in the patcher?

Reply:
e.g., would the dynamo tests still pass without this change now?

Reply:
beautiful :D though this code is still ??? to me 😛 

I'm thinking of revamping all optim tests to use my new OptimizerInfos so I will get to refactoring this down the road.